collections/soldiers.js

```diff
@@ -0,0 +1,33 @@
+const MongoClient = require('mongodb').MongoClient;
+
+const url = 'mongodb://localhost:27017';
+const dbName = 'myproject';
+const soldierFields = 5
+
+function checkData(soldierData) {
+  return (/tt[0-9]{7}/.test(soldierData["id"]) &&
+  soldierData["name"] != null &&
+  soldierData["rank"] != null &&
+  soldierData["limitations"] != null &&
+  soldierData["duties"] != null &&
+  Object.keys(soldierData).length === soldierFields)
```

Comment:
Add a parenthesis `((a)&&(b)&&...&&(f))`
It is done for extra safety and extra readability